pkgs/applications/graphics/vpype/default.nix

```diff
@@ -0,0 +1,68 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+
+, asteval
+, cachetools
+, click
+, multiprocess
+, numpy
+, pnoise
+, scipy
+, setuptools
+, shapely
+, svgelements
+, svgwrite
+, tomli
+
+# gui dependencies
+, glcontext
+, matplotlib
+, moderngl
+, pillow
+, pyside2
+
+, poetry-core
+}:
+
+buildPythonPackage rec {
+  pname = "vpype";
+  version = "1.10.0";
+
+  src = fetchFromGitHub {
+    # https://github.com/abey79/vpype
+    owner = "abey79";
+    repo = "vpype";
+    rev = version;
+    sha256 = "sha256-eecLBUVH4TCqaMPNvoMB7E6VNTtBLeoEcDsFEliv1Ig=";
+  };
+
+  format = "pyproject";
+
+  propagatedBuildInputs = [
+    cachetools click multiprocess numpy pnoise shapely scipy setuptools
+    svgelements svgwrite tomli asteval
+
+    matplotlib glcontext moderngl pillow pyside2
+
+    poetry-core
```

Comment:
`poetry-core` should maybe be passed in `nativeBuildInputs` as it is only required at build time?